calico_node/tests/st/policy/test_felix_gateway.py

```diff
@@ -140,216 +154,285 @@ def tearDownClass(cls):
         for host in cls.hosts:
             host.cleanup()
             del host
+        cls.calinet.delete()
 
         log_and_run("docker rm -f cali-st-ext-nginx || true")
 
-    def test_ingress_policy_can_block_through_traffic(self):
-        self.add_policy({
-            'apiVersion': 'v1',
-            'kind': 'policy',
-            'metadata': {'name': 'port80-int'},
-            'spec': {
-                'order': 10,
-                'ingress': [
-                    {
-                        'protocol': 'tcp',
-                        'destination': {'ports': [80]},
-                        'action': 'deny'
-                    },
-                ],
-                'egress': [
-                    {'action': 'deny'},
-                ],
-                'selector': 'role == "gateway-int"'
-            }
-        })
-        self.add_gateway_internal_iface()
-        retry_until_success(self.assert_host_can_not_curl_ext, 3)
+    def test_can_connect_by_default(self):
+        retry_until_success(self.assert_host_can_curl_local, 3)
+        retry_until_success(self.assert_gateway_can_curl_ext, 3)
+        retry_until_success(self.assert_host_can_curl_ext, 3)
+        retry_until_success(self.assert_hostwl_can_access_workload, 3)
+        retry_until_success(self.assert_workload_can_curl_ext, 3)
```

Comment:
You call this group of 5 checks quite a lot - maybe make them into a function and pass in 5 flags for connect/no-connect?